brushfire-core/src/main/scala/com/stripe/brushfire/Lazy.scala

```diff
@@ -0,0 +1,36 @@
+package com.stripe.brushfire
+
+import com.twitter.algebird.{Monoid, Semigroup}
+
+/**
+  * Wrapper class to allow expensive [[Semigroup]] operations on a [[Tree]] to be deferred until first use
+  */
+class Lazy[A](fn: () => A) extends Serializable {
```

Comment:
Agreed on getting this into Algebird. I'm actually kinda surprised it doesn't exist there already.